docs/products/clickhouse/concepts/service-architecture.rst

```diff
@@ -0,0 +1,93 @@
+
+Each Aiven for ClickHouse service is exposed as a single server URL pointing to all servers with connections going randomly to any of the servers. Aiven for ClickHouse is responsible for replicating the writes between the servers.
+For synchronizing critical low-volume information between servers, Aiven for ClickHouse relies on :ref:`ZooKeeper <zookeeper>`, which runs on each ClickHouse server.
+
+Coordinating services
+---------------------
+
+Each Aiven for Clickhouse node runs Clickhouse, ZooKeeper, and Astacus.
+
+.. _zookeeper:
+
+ZooKeeper
+'''''''''
+
+ZooKeeper is responsible for coordination and synchronization across the nodes: one process per node, accessible only from within the cluster.
+
+* By Clickhouse's replicated :ref:`database engine <replicated-database-engine>`, to replicate database changes across the cluster (CREATE, UPDATE or ALTER TABLE)
+* By Clickhouse's ReplicatedMergeTree :ref:`table engine <replicated-table-engine>`, to coordinate replicating table data across the cluster. Data itself is not written to ZK, it is transferred directly between Clickhouse servers
+* By Clickhouse for Replicated Access Entity Storage, this is storage of Users, Roles, Quotas, Row Policies for the whole cluster, in ZooKeeper so that all cluster nodes have the same view of these entities. This storage was developed at Aiven and is now part of the upstream Clickhouse.
+
+Astacus
+'''''''
+
+Astacus is an open-source project originated at Aiven that coordinates backups of cluster databases, including ClickHouse.
+
+Data architecture
+-----------------
+
+* Aiven for ClickHouse enforces full schema replication: all databases, tables, users & grants are the same on all nodes.
+* Aiven for ClickHouse enforces full data replication: table rows are the same on all nodes within a shard.
+* Astacus does most of the backup & restore operations.
+
+.. image:: /images/products/clickhouse/data-architecture.png
+   :width: 800px
+   :alt: Data architecture
+
+Engines: database and table
+---------------------------
+
+ClickHouse has engines in two flavors: table engines and database engines.
+
+* Database engine is responsible for manipulating tables and decides what happens when you try to list, create, or delete a table. It can also be used to restrict a database to specific table engine or to manage replication.
+* Table engine decides how to store data on a disk or how to enable reading data from outside the disk to expose it as a virtual table.
+
+.. _replicated-database-engine:
+
+Replicated database engine
+''''''''''''''''''''''''''
+
+The default ClickHouse database engine is the Atomic engine responsbile for createing tables on the disk and authorizing table engines to databases.
+
+Aiven for ClickHouse uses the replicated database engine, which is a variant of Atomic. With this engine variant, queries for creating, updating, or altering tables are replicated to all other servers using :ref:`ZooKeeper <zookeeper>`.
```

Comment:
```suggestion
Aiven for ClickHouse uses the Replicated database engine, which is a variant of Atomic. With this engine variant, queries for creating, updating, or altering tables are replicated to all other servers using :ref:`ZooKeeper <zookeeper>`.
```
Replicated is the "real" full name of the engine in this context, not just an adjective.